lib/smb2/client.rb

```diff
@@ -1,7 +1,21 @@
 require 'smb2/packet'
```

Comment:
You're using `Kernel.autoload` in `lib/smb2.rb`, so you shouldn't be using explicit `require` here as it will add an unnecessary scan of `$LOAD_FEATURES` when the constants are already loaded.